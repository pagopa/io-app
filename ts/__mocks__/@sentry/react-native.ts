export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
