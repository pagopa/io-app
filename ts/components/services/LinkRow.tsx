import * as React from "react";
import { StyleSheet } from "react-native";

import { TranslationKeys } from "../../../locales/locales";
import I18n from "../../i18n";
import { openWebUrl } from "../../utils/url";
import { showToast } from "../../utils/showToast";

import { Link } from "../core/typography/Link";
import ItemSeparatorComponent from "../ItemSeparatorComponent";

const styles = StyleSheet.create({
  link: {
    paddingVertical: 16
  },
  header: {
    flexDirection: "row",
    paddingVertical: 8
  },
  icon: {
    marginRight: 3.85
  }
});

type Props = {
  text: TranslationKeys;
  href: string;
};

const LinkRow = ({ text, href }: Props) => (
  <>
    <Link
      ellipsizeMode={"tail"}
      onPress={() =>
        openWebUrl(href, () => showToast(I18n.t("global.jserror.title")))
      }
      numberOfLines={1}
      style={styles.link}
    >
      {I18n.t(text)}
    </Link>
    <ItemSeparatorComponent noPadded />
  </>
);

export default LinkRow;
