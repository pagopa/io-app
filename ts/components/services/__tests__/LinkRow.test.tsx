import React from "react";
import { render } from "@testing-library/react-native";

import LinkRow from "../LinkRow";

describe("LinkRow component", () => {
  it("should match the snapshot", () => {
    const component = render(
      <LinkRow text={"global.media.email"} href={"mail.com"} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
