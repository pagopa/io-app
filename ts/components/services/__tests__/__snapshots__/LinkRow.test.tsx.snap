// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkRow component should match the snapshot 1`] = `
Array [
  <Text
    accessibilityRole="link"
    color="blue"
    defaultColor="blue"
    defaultWeight="SemiBold"
    ellipsizeMode="tail"
    font="TitilliumWeb"
    fontStyle={
      Object {
        "fontSize": 16,
        "textDecorationLine": "underline",
      }
    }
    numberOfLines={1}
    onPress={[Function]}
    style={
      Array [
        Object {
          "paddingVertical": 16,
        },
        Object {
          "fontSize": 16,
          "textDecorationLine": "underline",
        },
        Object {
          "color": "#0073E6",
          "fontFamily": "Titillium Web",
          "fontStyle": "normal",
          "fontWeight": "600",
        },
      ]
    }
    weight="SemiBold"
  >
    Email
  </Text>,
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#C9C9C9",
        },
        false,
        Object {
          "height": 0.5,
        },
      ]
    }
  />,
]
`;
